bin/update-wdtk-rejections

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# update-wdtk-rejections:
+#
+#  Run WDTK Rake tasks to generate requests to block at MTA layer and
+#  munge output to generate simple regexes suitable for use with Exim's
+#  nwildlsearch lookup.
+#
+
+WDTK_DIR=${1:-/data/vhost/www.whatdotheyknow.com/alaveteli}
+
+# Are we running as root?
+if [ "$UID" != "0" ]; then
+  echo "This script should be run as root, aborting." >&2
+  exit 1
+fi
+
+# Is WDTK available where we think it is?
+if [ ! -d "$WDTK_DIR" ]; then
+  echo "WDTK doesn't appear to be availle in ${WDTK_DIR}, aborting." >&2
+  exit 2
+fi
+
+check_rake () {
+  RAKE_TASK=$1
+  RAKE_EXIT_CODE=$2
+  if [ "${RAKE_EXIT_CODE}" != "0" ]; then
+    echo "Fatal: ${RAKE_TASK} rake task returned non-zero, aborting." >&2
+    exit 3
+  fi
+}
+
+RAKE_OUTPUT=$(mktemp)
+SORTED_OUTPUT=$(mktemp)
+
+# Update the list of requests to block
+su -l foi -c "cd ${WDTK_DIR} && bundle exec rake config_files:set_reject_incoming_at_mta REJECTED_THRESHOLD=3 AGE_IN_MONTHS=6 DRYRUN=0; exit $?" > /dev/null
+check_rake "set_reject_incoming_at_mta" $?
+
+# Export the list
+su -l foi -c "cd ${WDTK_DIR} && bundle exec rake config_files:generate_mta_rejection_list MTA=exim; exit $?" > ${RAKE_OUTPUT}
+check_rake "generate_mta_rejection_list" $?
+
+# Sort the list to ensure consistent comparisons.
+sort ${RAKE_OUTPUT} > ${SORTED_OUTPUT}
+
+# Add to Puppet, if necessary.
+cd /data/puppet
+/data/mysociety/bin/git-safe-to-checkout . master >/dev/null
+if [ "$?" == "0" ]; then
+  git checkout --quiet master
+  git fetch --quiet >/dev/null
+  git rebase --quiet origin/master
+  diff ${SORTED_OUTPUT} ./site/exim/files/etc/exim4/recipient-reject >/dev/null 2>&1
+  if [ "$?" != "0" ]; then
+    cp ${SORTED_OUTPUT} ./site/exim/files/etc/exim4/recipient-reject
+    git add ./site/exim/files/etc/exim4/recipient-reject
+    git commit -m "Exim: Update WDTK requests blocked at MTA"
```

Comment:
Would be good to name the script in the commit message so that its easy to trace the update back to the code that caused it.

Reply:
Good idea, done.